### PR TITLE
Add overloads the CreateNew method of FileSystemWatcherFactory

### PR DESCRIPTION
**Pull request by nathanpovo**
Fixes issue #577.

### CODE REVIEW

**Comment by nathanpovo** [comment]
Should the `FromPath` method 

https://github.com/System-IO-Abstractions/System.IO.Abstractions/blob/7337796a9ee7c37a25cbc869630f0b2b7cd9812b/System.IO.Abstractions/FileSystemWatcherFactory.cs#L16-L19

be marked as obsolete or removed?

**Comment by fgreinacher** [comment]
Thanks for your contribution 👏 

> Should the `FromPath` method
> 
> https://github.com/System-IO-Abstractions/System.IO.Abstractions/blob/7337796a9ee7c37a25cbc869630f0b2b7cd9812b/System.IO.Abstractions/FileSystemWatcherFactory.cs#L16-L19
> 
> be marked as obsolete or removed?

Yes, please remove it and bump the major version in https://github.com/System-IO-Abstractions/System.IO.Abstractions/blob/master/version.json#L3.

File: System.IO.Abstractions/FileSystemWatcherFactory.cs [hunk 1]
```diff
@@ -13,6 +8,16 @@ public IFileSystemWatcher CreateNew()
             return new FileSystemWatcherWrapper();
         }
 
+        public IFileSystemWatcher CreateNew(string path)
+        {
+            return new FileSystemWatcherWrapper(path);
+        }
```
Comment (fgreinacher) [inline System.IO.Abstractions/FileSystemWatcherFactory.cs:14]:
```suggestion
        public IFileSystemWatcher CreateNew(string path) => 
            new FileSystemWatcherWrapper(path);
```

File: System.IO.Abstractions/FileSystemWatcherFactory.cs [hunk 2]
```diff
@@ -13,6 +8,16 @@ public IFileSystemWatcher CreateNew()
...
+
+        public IFileSystemWatcher CreateNew(string path, string filter)
+        {
+            return new FileSystemWatcherWrapper(path, filter);
+        }
```
Comment (fgreinacher) [inline System.IO.Abstractions/FileSystemWatcherFactory.cs:19]:
```suggestion
        public IFileSystemWatcher CreateNew(string path, string filter)
            => new FileSystemWatcherWrapper(path, filter);
```

**Comment by fgreinacher** [comment]
Thank you!

**Pull request merged**